File: actividades/prog/py_vs_rb/A2/program1.py
import sys
import math

a = int(sys.argv[1])
b = int(sys.argv[2])
c = int(sys.argv[3])
discriminante = b ** 2 - 4 * a * c

print('Calculadora de Ecuaciones de Segundo Grado')
print('Coeficiente cuadrático (a) ---->', a)
print('Coeficiente lineal (b) ------->', b)
print('Término independiente (c) ----->', c)
print('Ecuación:',a,'x\xb2','+',b,'x','+',c)

input('Pulse ENTER para ver la solución')

if a == 0:
    x = -c / b
    print('Los valores introducidos no se corresponden con una ecuación de segundo grado')
    print('La solución es: x = ', x)

elif discriminante < 0:
    print('La ecuación no tiene solución real')

else:
    x1 = (-b + math.sqrt(discriminante)) / (2 * a)
    x2 = (-b - math.sqrt(discriminante)) / (2 * a)
    print('Soluciones: ')
    print('x1 = ', x1 )
    print('x2 = ', x2 )

File: explicaciones/ruby/py_vs_rb/a2/program1.rb
#!/usr/bin/ruby

a = ARGV[0].to_i
b = ARGV[1].to_i
c = ARGV[2].to_i
discriminante = b ** 2 - 4 * a * c

puts 'Calculadora de Ecuaciones de Segundo Grado'
puts "Coeficiente cuadrático (a) ----> #{a}"
puts "Coeficiente lineal (b) --------> #{b}"
puts "Término independiente (c) -----> #{c}"
puts "Ecuación: #{a} x^2 + #{b} x + #{c}"

puts('Pulse ENTER para ver la solución')
#p = gets

if a == 0
  x = -c / b
  puts 'Los valores introducidos no se corresponden con una ecuación de segundo grado'
  puts "La solución es: x = #{x}"
elsif discriminante < 0
  puts 'La ecuación no tiene solución real'
else
  x1 = (-b + Math.sqrt(discriminante.to_f)) / (2 * a)
  x2 = (-b - Math.sqrt(discriminante.to_f)) / (2 * a)
  puts 'Soluciones: '
  puts "x1 = #{x1}"
  puts "x2 = #{x2}"
end
